### ISSUE TITLE
ResNet18 exported with dynamo=True gives zero accuracy on ImageNet-1K

### ISSUE THREAD

**Issue by andrey-churkin** [pytorch/pytorch#169178]
### 🐛 Describe the bug

When exporting the `ResNet18` model from `torchvision` with `dynamo=True`, the resulting model produces 0% accuracy on the ImageNet-1K validation set. Exporting the same model with `dynamo=False` works correctly, and the model achieves expected accuracy.

**Environment:**
torch: 2.9.1+cu128
torchvision: 0.24.1+cu128
onnx: 1.17.0
onnxruntime: 1.21.1
onnxscript: 0.5.6

**Reproducer:**
```python
import torch
import torchvision
from torchvision.models import resnet18, ResNet18_Weights
from sklearn.metrics import accuracy_score
import onnx
import onnxruntime
import onnxscript
import numpy as np
from pathlib import Path
from tqdm import tqdm

def export(dynamo: bool, model_path: str):
    model = resnet18(weights=ResNet18_Weights.DEFAULT)
    model.eval()
    preprocess = ResNet18_Weights.DEFAULT.transforms()

    batch_size = 128
    dummy_input = torch.rand(batch_size, 3, 224, 224)
    dummy_input = preprocess(dummy_input)

    additional_kwargs = {}
    if dynamo:
        additional_kwargs["dynamic_shapes"] =({0: "batch_size"},)
    else:
        additional_kwargs["dynamic_axes"] = {"image": {0: "batch"}}

    if dynamo:
        onnx_program = torch.onnx.export(
             model,
             dummy_input,
             export_params=True,
             opset_version=18,
             dynamo=True,
             input_names=["image"],
             output_names=["predictions"],
             **additional_kwargs,
        )
        onnx_program.save(model_path)
    else:
        torch.onnx.export(
            model,
            dummy_input,
            model_path,
            export_params=True,
            opset_version=18,
            dynamo=False,
            input_names=["image"],
             output_names=["predictions"],
            **additional_kwargs,
        )

def validate(onnx_model_path: str, data_path: Path):
    dataset = torchvision.datasets.ImageFolder(
        data_path / "imagenet" / "val",
        ResNet18_Weights.DEFAULT.transforms()
    )
    data_loader = torch.utils.data.DataLoader(dataset, batch_size=1, shuffle=False)

    sess = onnxruntime.InferenceSession(onnx_model_path, providers=["CPUExecutionProvider"])

    dataset_size = len(data_loader)
    print(f"Dataset Size: {dataset_size}")
    predictions = np.zeros(dataset_size)
    references = -1 * np.ones(dataset_size)

    for i, (images, target) in tqdm(enumerate(data_loader), total=dataset_size, desc="Validation"):
        output = sess.run([], {"image": images.numpy()})[0]
        predictions[i] = np.argmax(output, axis=1).item()
        references[i] = target.item()

    acc_top1 = accuracy_score(predictions, references)
    return acc_top1

if __name__ == "__main__":
    RESNET18_DYNAMO_FALSE = "resnet18_dynamo_false.onnx"
    RESNET18_DYNAMO_TRUE = "resnet18_dynamo_true.onnx"
    DATASET_PATH = Path("<YOUR PATH HERE>")

    print("========== Export ==========")
    export(dynamo=False, model_path=RESNET18_DYNAMO_FALSE)
    export(dynamo=True, model_path=RESNET18_DYNAMO_TRUE)

    print("========== Validation ==========")
    print(f"Model: {RESNET18_DYNAMO_FALSE}")
    acc_top1 = validate(RESNET18_DYNAMO_FALSE, DATASET_PATH)
    print(f"{RESNET18_DYNAMO_FALSE}: acc@1={acc_top1}")

    print(f"Model: {RESNET18_DYNAMO_TRUE}")
    acc_top1 = validate(RESNET18_DYNAMO_TRUE, DATASET_PATH)
    print(f"{RESNET18_DYNAMO_TRUE}: acc@1={acc_top1}")
```

**Reproducer Output**:
```text
Model: resnet18_dynamo_false.onnx
Dataset Size: 50000
Validation: 100%
resnet18_dynamo_false.onnx: acc@1=0.6976
Model: resnet18_dynamo_true.onnx
Dataset Size: 50000
Validation: 100%
resnet18_dynamo_true.onnx: acc@1=0.00376
```

### Versions

Collecting environment information...
PyTorch version: 2.9.1+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: version 3.26.0
Libc version: glibc-2.31

Python version: 3.10.17 (main, Nov 26 2025, 08:09:25) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-117-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: 
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090
GPU 2: NVIDIA GeForce RTX 3090

Nvidia driver version: 550.54.14
cuDNN version: Probably one of the following:
/usr/local/cuda-12.2/targets/x86_64-linux/lib/libcudnn.so.8
/usr/local/cuda-12.2/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8
/usr/local/cuda-12.2/targets/x86_64-linux/lib/libcudnn_adv_train.so.8
/usr/local/cuda-12.2/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8
/usr/local/cuda-12.2/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8
/usr/local/cuda-12.2/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8
/usr/local/cuda-12.2/targets/x86_64-linux/lib/libcudnn_ops_train.so.8
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Byte Order:                           Little Endian
Address sizes:                        46 bits physical, 48 bits virtual
CPU(s):                               36
On-line CPU(s) list:                  0-35
Thread(s) per core:                   2
Core(s) per socket:                   18
Socket(s):                            1
NUMA node(s):                         1
Vendor ID:                            GenuineIntel
CPU family:                           6
Model:                                85
Model name:                           Intel(R) Core(TM) i9-10980XE CPU @ 3.00GHz
Stepping:                             7
CPU MHz:                              3830.694
CPU max MHz:                          4800.0000
CPU min MHz:                          1200.0000
BogoMIPS:                             6000.00
Virtualization:                       VT-x
L1d cache:                            576 KiB
L1i cache:                            576 KiB
L2 cache:                             18 MiB
L3 cache:                             24.8 MiB
NUMA node0 CPU(s):                    0-35
Vulnerability Gather data sampling:   Mitigation; Microcode
Vulnerability Itlb multihit:          KVM: Mitigation: VMX disabled
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Mitigation; Enhanced IBRS
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Mitigation; TSX disabled
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512_vnni md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] numpy==2.0.2
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-nccl-cu12==2.27.5
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] onnx==1.17.0
[pip3] onnx-ir==0.1.12
[pip3] onnxruntime==1.21.1
[pip3] onnxscript==0.5.6
[pip3] optimum-onnx==0.0.3
[pip3] torch==2.9.1
[pip3] torchvision==0.24.1
[pip3] triton==3.5.1
[conda] Could not collect

cc @justinchuby @titaiwangms @chauhang @penguinwu @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

**Comment by xadupre** [comment]
Is it possible to try with the nightly version of pytorch and the latest onnxruntime? I was not able to replicate your issue. It is possible to share an image for which you see discrepancies?

**Comment by titaiwangms** [comment]
Please try nightly
